### PR TITLE
Filter update

### PR DESCRIPTION
**Pull request by reinaldoferreira**

### CODE REVIEW

File: src/js/typeahead-filter.js [hunk 1]
```diff
@@ -1,180 +1,172 @@
...
+  let suggestionsFilter = new Suggestions(el, data)
+  suggestionsFilter.getItemValue = item => item.Label
+}
+
+// Filter related
+const CCG_ENDPOINT = `/umbraco/api/ccg/labels?nation=${DataNation}`
+const CANCER_ENDPOINT = `/umbraco/api/cancer/labels?nation=${DataNation}`
+
+const formData = (result, obj) => {
+  let mergedObj = Object.assign(result, obj, {})
```
Comment (richard-cargill) [inline src/js/typeahead-filter.js]: var mergedObj = $.extend({}, result, obj);

File: src/js/typeahead-filter.js [hunk 2]
```diff
@@ -1,180 +1,172 @@
...
+
+// Run code on each element if there are any elements
+Array.prototype.map.call(elems, el => {
+  let req1 = request(CCG_ENDPOINT)
+  let req2 = request(CANCER_ENDPOINT)
+  $.when(req1, req2).done((res1, res2) => {
+    let res1Data = formData(res1[0], { type: 'ccg' })
+    let res2Data = formData(res2[0], { type: 'cancer' })
+    let data = Object.assign(res1Data, res2Data, {})
+
```
Comment (richard-cargill) [inline src/js/typeahead-filter.js]: var mergedObj = $.extend({}, result, obj);

File: src/js/typeahead-filter.js [hunk 3]
```diff
@@ -1,180 +1,172 @@
...
+// Impure functions
+const toggleBtnState = (state = true) => {
+  let toggleState = !state ? 'removeAttribute' : 'setAttribute'
+  $(btnUpdateFilter).toggleClass('is--disabled', state)
+  btnUpdateFilter[toggleState]('disabled', '')
+}
+
+const validateFilter = (obj) => {
+  let searchValues = Object.values(obj)
+
```
Comment (richard-cargill) [inline src/js/typeahead-filter.js]: let searchValues = [obj.ccg, obj.cancer]

File: src/js/typeahead-filter.js (hunk 1 above)
Comment (reinaldoferreira) [inline src/js/typeahead-filter.js]: instead of $.extend I had to go with $.merge because I need to filter an array

**Pull request merged**